webapi/src/test/scala/org/knora/webapi/models/FileModels.scala

```diff
@@ -0,0 +1,1131 @@
+  val stillImageRepresentation = "StillImageRepresentation"
+
+  /** "MovingImageRepresentation" */
+  val movingImageRepresentation = "MovingImageRepresentation"
+
+  /** "AudioRepresentation" */
+  val audioRepresentation = "AudioRepresentation"
+
+  /** "BundleRepresentation" */
+  val bundleRepresentation = "BundleRepresentation"
+
+  /** "knora-api" */
+  val knoraApiPrefix = "knora-api"
+
+  /** "0001" */
+  val anythingShortcode = "0001"
+}
+
+/**
+ * Models for generating JSON-LD requests to manipulate file representations.
+ */
+object FileJsonModels {
+  import FileModelConstants._
+
+  /**
+   * Trait for FileValue type.
+   *
+   * Implemented types:
+   *
+   *  - knora-api:DocumentFileValue
+   *  - knora-api:StillImageFileValue
+   *  - knora-api:TextFileValue
+   *  - knora-api:TextFileValue
+   *  - knora-api:BundleRepresentation
+   */
+  sealed trait FileValueType {
+
+    /** String value of the FileValueType */
+    val value: String
+  }
+
+  /**
+   * Companion object to the [[FileValueType]] trait.
+   */
+  object FileValueType {
+
+    /**
+     * Represents "knora-api:DocumentFileValue"
+     */
+    case object DocumentFileValue extends FileValueType {
+      val value = "knora-api:DocumentFileValue"
+    }
+
+    /**
+     * Represents "knora-api:StillImageFileValue"
+     */
+    case object StillImageFileValue extends FileValueType {
+      val value = "knora-api:StillImageFileValue"
+    }
+
+    /**
+     * Represents "knora-api:MovingImageFileValue"
+     */
+    case object MovingImageFileValue extends FileValueType {
+      val value = "knora-api:MovingImageFileValue"
+    }
+
+    /**
+     * Represents "knora-api:TextFileValue"
+     */
+    case object TextFileValue extends FileValueType {
+      val value = "knora-api:TextFileValue"
+    }
+
+    /**
+     * Represents "knora-api:AudioFileValue"
+     */
+    case object AudioFileValue extends FileValueType {
+      val value = "knora-api:AudioFileValue"
+    }
+
+    /**
+     * Represents "knora-api:BundleFileValue"
+     */
+    case object BundleFileValue extends FileValueType {
+      val value = "knora-api:BundleFileValue"
+    }
+  }
+
+  /**
+   * Case class containing the JSON-LD serialization of a request to create a "knora-api:DocumentRepresentation".
+   *
+   * @param value the JSON-LD serialized request.
+   */
+  sealed abstract case class UploadDocumentFile private (value: String)
+
+  /**
+   * Companion object to the [[UploadDocumentFile]] case class.
+   */
+  object UploadDocumentFile {
+
+    /**
+     * Smart constructor for instantiating a [[UploadDocumentFile]].
+     *
+     * @param internalFilename the internal file name assigned by SIPI.
+     * @param className the class name of the resource. Defaults to "DocumentRepresentation".
+     *                  Can be a project ontology class that inherits from "DocumentRepresentation".
+     * @param shortcode the shortcode of the project to which the resource should be added.
```

Comment:
Shouldn't be `shortCode` (also in everywhere below)?

Reply:
generally yes, but in our code base and docs, shortcode is never in camel case, so for consistency I'd rather leave it here like this too